### ISSUE TITLE
AWS partition not set correctly

### ISSUE THREAD

**Issue by ely-alamillo** [anomalyco/sst#5076]
I'm following guide https://sst.dev/docs/start/aws/nextjs#containers to deploy a next application. I have created an iam user with admin access for this. When running sst dev I always get this error
```
  aws: failed to put bucket policy for sst-asset-ttkxozvttknn: operation error S3: PutBucketPolicy, https response error StatusCode: 400, RequestID: QMHE309T3Z4WTSTZ, HostID: LVRhJQflRFlTCt9MYA4tIbBgsYw4vP1uccsLtGqSY88lP6kyjisbGZA9wWjlldAN/AIIq+zbPCg=, api error MalformedPolicy: Policy has invalid resource
```
i'm trying to deploy this to us-gov-west-1, i tried to do the same on a diff acount for us-east-1 and it seems to work fine. 

alright further down the rabbit hole [this step](https://github.com/sst/sst/blob/b58f34dc86751423305fe3ac666ae16effe20c00/pkg/project/provider/aws.go#L407) generates this bucket policy
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "EnforceSSLRequests",
      "Effect": "Deny",
      "Principal": "*",
      "Action": "s3:*",
      "Resource": [
        "arn:aws:s3:::sst-asset-fumdarbaxwao",
        "arn:aws:s3:::sst-asset-fumdarbaxwao/*"
      ],
      "Condition": {
        "Bool": {
          "aws:SecureTransport": "false"
        }
      }
    }
  ]
}
```

the failing part is the Resource which is set to be arn:aws:s3:::sst-asset-fumdarbaxwao but in govcloud regions it should be arn:aws-us-gov:s3:::sst-asset-fumdarbaxwao

I took a look at the code that generates this config and the issue is that the aws partition is hard coded to `aws` which isn't correct for partitions such as govcloud.

**Comment by jayair** [comment]
Interesting, let me tell the team.

**Comment by fwang** [comment]
Fixed in v3.3.35 /cc @ely-alamillo
